src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts

```diff
@@ -505,6 +510,11 @@ class TerminalTabsRenderer extends Disposable implements IListRenderer<ITerminal
 				this._runForSelectionOrInstance(instance, e => this._terminalService.safeDisposeTerminal(e));
 			}))
 		];
+		if (instance.shellLaunchConfig.type === 'Task') {
+			actions.push(this._register(new Action(TerminalCommandId.RerunTask, terminalStrings.rerunTask.value, ThemeIcon.asClassName(Codicon.debugRestart), true, async () => {
+				this._runForSelectionOrInstance(instance, e => this._commandService.executeCommand(TerminalCommandId.RerunTask, instance));
+			})));
+		}
```

Comment:
We want this all to live within `tasks/`, so the changes in `terminal/` should just be about exposing APIs that tasks can work with